src/test/java/igrad/model/course/CourseInfoTest.java

```diff
@@ -410,28 +431,45 @@ public void equals() {
         Module module = new ModuleBuilder().build();
         assertFalse(BCOMPSCI.equals(module));
 
+        assertFalse(BCOMPSCI.equals(GENERAL_ELECTIVES));
+
         CourseInfo otherCourseInfo;
 
         // different course info; only cap different
         otherCourseInfo = new CourseInfoBuilder(BCOMPSCI)
             .withCap(VALID_COURSE_CAP_BCOMPSEC)
             .build();
-        assertFalse(GENERAL_ELECTIVES.equals(otherCourseInfo));
+        assertFalse(BCOMPSCI.equals(otherCourseInfo));
 
         // different course info; only credits different
         otherCourseInfo = new CourseInfoBuilder(BCOMPSCI)
             .withCredits(VALID_COURSE_CREDITS_REQUIRED_BCOMPSEC,
                 VALID_COURSE_CREDITS_FULFILLED_BCOMPSEC)
             .build();
-        assertFalse(GENERAL_ELECTIVES.equals(otherCourseInfo));
+        assertFalse(BCOMPSCI.equals(otherCourseInfo));
+
+        //different course info; only semesters different
+        otherCourseInfo = new CourseInfoBuilder(BCOMPSCI)
+                .withSemesters(VALID_COURSE_SEMESTERS_BCOMPSEC)
+                .build();
+        assertFalse(BCOMPSCI.equals(otherCourseInfo));
 
         // different course info; both cap and credits, different
         CourseInfo other = new CourseInfoBuilder(BCOMPSCI)
             .withCap(VALID_COURSE_CAP_BCOMPSEC)
             .withCredits(VALID_COURSE_CREDITS_REQUIRED_BCOMPSEC,
                 VALID_COURSE_CREDITS_FULFILLED_BCOMPSEC)
             .build();
-        assertFalse(GENERAL_ELECTIVES.equals(other));
+        assertFalse(BCOMPSCI.equals(other));
+
+        //different course info; cap, credits, semesters, all different
+        other = new CourseInfoBuilder(BCOMPSCI)
```

Comment:
good but you could have added another case  above, where only the `semesters` was different

Reply:
I did, its slightly above